.cirrus.yml

```diff
@@ -0,0 +1,115 @@
+  PROMOTE_URL: ENCRYPTED[!e22ed2e34a8f7a1aea5cff653585429bbd3d5151e7201022140218f9c5d620069ec2388f14f83971e3fd726215bc0f5e!]
+
+  GITHUB_TOKEN: ENCRYPTED[!f272985ea5b49b3cf9c414b98de6a8e9096be47bfcee52f33311ba3131a2af637c1b956f49585b7757dd84b7c030233a!]
+
+  BURGR_URL: ENCRYPTED[!c7e294da94762d7bac144abef6310c5db300c95979daed4454ca977776bfd5edeb557e1237e3aa8ed722336243af2d78!]
+  BURGR_USERNAME: ENCRYPTED[!b29ddc7610116de511e74bec9a93ad9b8a20ac217a0852e94a96d0066e6e822b95e7bc1fe152afb707f16b70605fddd3!]
+  BURGR_PASSWORD: ENCRYPTED[!83e130718e92b8c9de7c5226355f730e55fb46e45869149a9223e724bb99656878ef9684c5f8cfef434aa716e87f4cf2!]
+
+  ### Project variables
+  DEPLOY_PULL_REQUEST: true
+  ARTIFACTS: org.sonarsource.scanner.maven:sonar-maven-plugin:jar
+
+#
+# RE-USABLE CONFIGS
+#
+container_definition: &CONTAINER_DEFINITION
+  image: gcr.io/ci-cd-215716/base:latest
+  cluster_name: cirrus-uscentral1a-cluster
+  zone: us-central1-a
+  namespace: default
+
+only_sonarsource_qa: &ONLY_SONARSOURCE_QA
+  only_if: $CIRRUS_USER_COLLABORATOR == 'true' && ($CIRRUS_PR != "" || $CIRRUS_BRANCH == "master" || $CIRRUS_BRANCH =~ "branch-.*" || $CIRRUS_BRANCH =~ "dogfood-on-.*")
+
+#
+# TASKS
+#
+build_task:
+  gke_container:
+    <<: *CONTAINER_DEFINITION
+    cpu: 2
+    memory: 2G
+  env:
+    SONAR_TOKEN: ENCRYPTED[!5ba7cbb5bf9d168de69bcd444d9e884c9cf664be1115640cc64e49df6d241c309a87fc527cab533c08f289b167187017!]
+    SONAR_HOST_URL: https://next.sonarqube.com/sonarqube
+  maven_cache:
+    folder: ${CIRRUS_WORKING_DIR}/.m2/repository
+  script:
+    - source cirrus-env BUILD
+    - regular_mvn_build_deploy_analyze
+  cleanup_before_cache_script:
+    - cleanup_maven_repository
+
+qa_task:
+  depends_on:
+    - build
+  <<: *ONLY_SONARSOURCE_QA
+  gke_container:
+    dockerfile: its/docker/Dockerfile
+    builder_image_project: ci-cd-215716
+    builder_image_name: docker-builder-v2
+    cluster_name: cirrus-uscentral1a-cluster
+    zone: us-central1-a
+    namespace: default
+    cpu: 1.7
+    memory: 5Gb
+  env:
+    matrix:
+      - SQ_VERSION: LATEST_RELEASE[7.9]
+        MAVEN_VERSION: 3.6.1
+      - SQ_VERSION: LATEST_RELEASE[7.9]
+        MAVEN_VERSION: 3.5.2
+      - SQ_VERSION: LATEST_RELEASE[7.9]
+        MAVEN_VERSION: 3.3.9
+      - SQ_VERSION: LATEST_RELEASE[7.9]
+        MAVEN_VERSION: 3.2.5
+      - SQ_VERSION: LATEST_RELEASE[7.9]
+        MAVEN_VERSION: 3.1.1
+      - SQ_VERSION: LATEST_RELEASE[7.9]
+        MAVEN_VERSION: 3.0.5
+      - SQ_VERSION: DEV
+        MAVEN_VERSION: 3.6.1
+  maven_cache:
+    folder: ${CIRRUS_WORKING_DIR}/.m2/repository
+  qa_script:
+    - source cirrus-env QA
+    - source set_maven_build_version $BUILD_NUMBER
+    - cd its
+    - curl -sSL https://repo1.maven.org/maven2/org/apache/maven/apache-maven/$MAVEN_VERSION/apache-maven-$MAVEN_VERSION-bin.tar.gz | tar zx --strip-components 1 -C $MAVEN_HOME_IT
+    - cp -f ~/.m2/settings.xml $MAVEN_HOME_IT/conf/
```

Comment:
It could be interesting to :
- Move all these script in a dedicated file script, like this for instance https://github.com/SonarSource/sonar-db-copy/blob/master/cirrus/cirrus-qa.sh
- Add some comments to explain why these operations are done

Reply:
Yeah, you're probably right. +💯for the comments!